design/automated-provisioning-design.md

```diff
@@ -0,0 +1,147 @@
+
+Again, the Gateway provisioner will be an optional component; users who are not interested in Gateway API or dynamic provisioning of Gateways are not required to use it.
+Those users can continue to statically provision their Contour + Envoy instances, for use with either Gateway API or traditional Ingress / HTTPProxy.
+
+There will be two ways to configure Contour for Gateway API support in the static provisioning scenario:
+- **Controller name** - this is the model implemented today, where Contour is configured with a controller name, and it continuously looks for the oldest GatewayClass with that controller, and the oldest Gateway using that GatewayClass. This model is appropriate for users who expect their GatewayClasses and Gateways to come and go, and who want their Contour instance to dynamically pick up the appropriate Gateway as those changes occur.
+- **Gateway name** - Contour can alternately be directly configured with a specific Gateway name, which avoids the multiple levels of indirection of the previous model. This model is appropriate for users who expect their Contour instance to correspond to a single static Gateway; the lifecycle of the Gateway and the lifecycle of the Contour instance are tied together.
+
+### GatewayClass Parameters
+
+Drawing on [design work already done](https://github.com/projectcontour/contour/blob/main/design/configuration-crd.md), the Gateway provisioner’s GatewayClasses will support configuration parameter references to ContourDeployment custom resources.
+This custom resource definition embeds a ContourConfiguration spec, as well as additional fields that control the precise details of the Contour + Envoy resources’ YAML specs (e.g. replicas, node placement, etc.)
+This ContourDeployment resource serves as a template that defines exactly how to customize each Contour + Envoy instance that is created for this GatewayClass.
+When a Gateway is provisioned, the Gateway provisioner will use the configuration options specified to customize the YAML that is applied, and will pass through a copy of the ContourConfiguration data to the Gateway’s Contour instance.
+
+Note that, according to the Gateway API documentation: 
+> It is recommended that [GatewayClass] be used as a template for Gateways.
+> This means that a Gateway is based on the state of the GatewayClass at the time it was created and changes to the GatewayClass or associated parameters are not propagated down to existing Gateways.
+> This recommendation is intended to limit the blast radius of changes to GatewayClass or associated parameters. 
+(ref. [GatewayClass API reference documentation](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.GatewayClass)) 
+
+This also implies that Contour users can modify the ContourConfigurations used by their running Contours after instantiation, without having those changes overwritten by the Gateway provisioner. 
```

Comment:
To put it another way, when a Gateway is created, the provisioner will create a ContourConfiguration resource for that deployment as described in the ContourDeployment resource, but after that won't touch it?